rdmd.d

```diff
@@ -916,3 +895,58 @@ void yap(size_t line = __LINE__, T...)(auto ref T stuff)
     debug stderr.writeln(line, ": ", stuff);
     else stderr.writeln(stuff);
 }
+
+//
+// std.file wrappers that also log their operations
+//
+auto exists(R)(R name)
+{
+    yap("stat ", name);
+    return std.file.exists(name);
+}
+auto isFile(R)(R name)
+{
+    yap("stat ", name);
+    return std.file.isFile(name);
+}
+auto isDir(R)(R name)
+{
+    yap("stat ", name);
+    return std.file.isDir(name);
+}
+auto existsAsFile(R)(R name)
+{
+    yap("stat ", name);
+    return std.file.exists(name) && std.file.isFile(name);
+}
+auto timeLastModified(R)(R name)
+{
+    yap("stat ", name);
+    return std.file.timeLastModified(name);
+}
+auto timeLastModified(R)(R name, SysTime returnIfMissing)
+{
+    yap("stat ", name);
+    return std.file.timeLastModified(name, returnIfMissing);
+}
+void remove(R)(R name)
+{
+    yap("rm ", name);
+    std.file.remove(name);
+}
+void rename(R,S)(R from, S to)
+{
+    yap("mv ", from, " ", to);
+    std.file.rename(from, to);
+}
+void mkdirRecurseIfLive(R)(R name)
+{
+    yap("mkdirRecurse ", name);
+    if (!dryRun)
+        return std.file.mkdirRecurse(name);
```

Comment:
This touches on a rather interesting problem of `rdmd`: dry-run mode is implemented manually throughout the codebase.  This means that in various places `if (!dryRun)` blocks mean that different output will actually be produced for `--dry-run` as opposed to just `--chatty`.

It might therefore be worth rethinking the wrapper functions as not just encapsulating the needed `yap` calls but also encapsulating the `if (!dryRun)` as and where needed (i.e. for those filesystem calls that actually _change_ the filesystem, as opposed to just querying it).

Reply:
this is the only function affected by dryRun.  this approach makes sense to me